protocol.md: Fix note formatting and add a note about annotations and image lists (#6)
owtaylor/metastore: docs/protocol.md

- `annotation:<annotation>=<value>`, `annotation:<annotation>:exists=1`: Limit results to images where the annotation `<annotation>` has the given value, or for the `:exists` form, exists with any value.
- `label:<label>=<value>`, `label:<label>:exists=1`: Limit results to images where the label `<label>` has the given value, or for the `:exists` form, exists with any value.

For parameters with `:exists` suffix, it is undefined what happens if any value other than `1` is specified.

Mutliple query parameters limit results to images that satisfy all of them
simultaneously. (So e.g., `?repository=r1&repository=r2` will always return

> empty results.)

Reviewer: This means that it will be interpreted as an AND condition, which is wrong. The correct interpretation is OR: https://github.com/pulp/pulp_container/pull/1301#discussion_r1238144555.

Author: Yeah, that was the old, wrong understanding from before the recent clarifications.  Addressed here now with updated <https://github.com/owtaylor/flagstate/pull/6/commits/a0e3b3086db37044cd9052b417ed6fea80569e5f> "Clarify how query parameters are matched".

Notes
-----

> * For images within an image list, the architecture matched by `architecture=` queries and returned in the JSON result is the architecture extracted from the images `config.json`, not the architecture in the manifest list or image index.

Reviewer: ```suggestion
* For images within a manifest list or OCI index, the architecture matched by `architecture=` queries and returned in the JSON result is the architecture extracted from the images `config.json`.
```

The architecture attribute must be specified in respective listed manifests: https://docs.docker.com/engine/reference/commandline/manifest/#inspect-a-manifest-list, https://github.com/opencontainers/image-spec/blob/main/image-index.md#image-index-property-descriptions. What is the rationale behind this statement, @owtaylor?

Reviewer: Oh, my bad, the platform object that holds the architecture attribute is optional.
